src/main/java/com/kaziamyr/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.kaziamyr.onlinebookstore.service.impl;
+
+import com.kaziamyr.onlinebookstore.dto.UserRegistrationRequestDto;
+import com.kaziamyr.onlinebookstore.dto.UserRegistrationResponseDto;
+import com.kaziamyr.onlinebookstore.exception.RegistrationException;
+import com.kaziamyr.onlinebookstore.mapper.UserMapper;
+import com.kaziamyr.onlinebookstore.model.Role;
+import com.kaziamyr.onlinebookstore.model.User;
+import com.kaziamyr.onlinebookstore.repository.UserRepository;
+import com.kaziamyr.onlinebookstore.service.UserService;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserRegistrationResponseDto register(UserRegistrationRequestDto request)
+            throws RegistrationException {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
+            throw new RegistrationException("User with email "
+                    + request.getEmail() + " is already present!");
+        }
+        User user = new User();
+        user.setPassword(passwordEncoder.encode(request.getPassword()));
```

Comment:
optional: actually it is better to create separate mapper class for such purposes